tests/priority-queue.spec.ts

```diff
@@ -0,0 +1,599 @@
+			expect(spy).toBeCalledTimes(0);
+			expect(instance.elements).toStrictEqual([10, 20, 60, 30, 50, 80, 70, 90, 40, 100]);
+		});
+
+		it('should do # swapNodes calls and be in heap order after adding new lowest rank to beginning', () => {
+			const spy = jest.spyOn(instance, 'swapNodes');
+
+			items.forEach((item) => {
+				instance.push(item);
+			});
+			spy.mockClear();
+
+			expect(instance.size()).toBe(items.length);
+			instance.elements[0] = 1;
+			// [1, 20, 60, 30, 50, 80, 70, 90, 40] --> [1, 20, 60, 30, 50, 80, 70, 90, 40]
+			// 0 swapNode calls to fix
+			instance.fixHeap(0);
+			expect(spy).toBeCalledTimes(0);
+			expect(instance.elements).toStrictEqual([1, 20, 60, 30, 50, 80, 70, 90, 40]);
+		});
+
+		it('should do # swapNodes calls and be in heap order after adding middle value rank to beginning', () => {
+			const spy = jest.spyOn(instance, 'swapNodes');
+
+			items.forEach((item) => {
+				instance.push(item);
+			});
+			spy.mockClear();
+
+			expect(instance.size()).toBe(items.length);
+			instance.elements[0] = 35;
+			// [55, 20, 60, 30, 50, 80, 70, 90, 40] --> [20, 30, 60, 40, 50, 80, 70, 90, 55]
+			// 2 swapNode calls to fix
+			instance.fixHeap(0);
+			expect(spy).toBeCalledTimes(2);
+			expect(instance.elements).toStrictEqual([20, 30, 60, 35, 50, 80, 70, 90, 40]);
+		});
+
+		it('should do # swapNodes calls and be in heap order after adding new highest rank to beginning', () => {
+			const spy = jest.spyOn(instance, 'swapNodes');
+
+			items.forEach((item) => {
+				instance.push(item);
+			});
+			spy.mockClear();
+
+			expect(instance.size()).toBe(items.length);
+			instance.elements[0] = 99;
+			// [99, 20, 60, 30, 50, 80, 70, 90, 40] --> [20, 30, 60, 40, 50, 80, 70, 90, 99]
+			// 3 swapNode calls to fix
+			instance.fixHeap(0);
+			expect(spy).toBeCalledTimes(3);
+			expect(instance.elements).toStrictEqual([20, 30, 60, 40, 50, 80, 70, 90, 99]);
+		});
+	});
+
+	describe('push', () => {
+		it('should add exactly one item to priority queue when push is called once', () => {
+			expect(instance.size()).toBe(0);
+			instance.push(1);
+			expect(instance.size()).toBe(1);
+		});
+
+		it('should add exactly 15 items to priority queue when push is called 15 times', () => {
+			expect(instance.size()).toBe(0);
+
+			const limit = 15;
+			for (let i = 0; i < limit; i++) {
+				instance.push(Math.floor(Math.random() * 999));
+			}
+
+			expect(instance.size()).toBe(limit);
+		});
+	});
+
+	describe('pop', () => {
+		it('should remove exactly 1 item from the priority queue when pop is called once', () => {
+			const limit = 12;
+			for (let i = 0; i < limit; i++) {
+				instance.push(Math.floor(Math.random() * 999));
+			}
+			expect(instance.size()).toBe(limit);
+			instance.pop();
+			expect(instance.size()).toBe(limit - 1);
+		});
+
+		it('should return null when called on a priority queue of size 0', () => {
+			expect(instance.size()).toBe(0);
+			expect(instance.pop()).toBeNull();
+		});
+
```

Comment:
Great tests here. 1 extra test case: Check that the list with 1 elements returns the element, and then null when called twice. Then one more check when list of size 5 or (some arbitrary size) is called size + 1 times, it returns n elements + 1 null.